CMakeLists.txt

```diff
@@ -113,6 +113,24 @@ if ( "$ENV{CI}" AND "$ENV{APPVEYOR}" )
 endif()
 # HLSL Change Ends
 
+if(WIN32)
+  file(TO_NATIVE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/utils/hct/hcteolcheck.cmd HCTEOLCHECK)
+  add_custom_target(git_eol_check
+    ALL
+    COMMENT Running hcteolcheck.cmd - fails if i/crlf or i/mixed in index
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    COMMAND ${HCTEOLCHECK}
+    VERBATIM)
+else()
+  add_custom_target(git_eol_check
+    ALL
+    COMMENT Running EOL check - fails if i/crlf or i/mixed in index
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    COMMAND ! git ls-files --eol | grep -e "i/crlf" -e "i/mixed"
```

Comment:
We should probably run this as `sh -c '! git ls-files --eol | grep -e "i/crlf" -e "i/mixed"'`, that way weirdos like me who have non-posix compliant default shells won't get broken by this :)

Reply:
Great suggestion, thanks!